core/include/usModule.h

```diff
@@ -1,368 +1,430 @@
+   *    specified, is called. If the BundleActivator is invalid or throws an exception then:
+   *    - A bundle event of type BundleEvent::STOPPING is fired.
+   *    - %Any services registered by this bundle must be unregistered.
+   *    - %Any services used by this bundle must be released.
+   *    - %Any listeners registered by this bundle must be removed.
+   *    - A bundle event of type BundleEvent::STOPPED is fired.
+   *    - A std::runtime_error exception is then thrown.
+   * -# A bundle event of type BundleEvent::STARTED is fired.
+   *
+   * @throws std::runtime_error If this bundle could not be started.
+   */
+  virtual void Start();
+
+  /**
+   * Stop this bundle.
+   *
+   * The following steps are required to stop a bundle:
+   * -# If this bundle is in the process of being activated or deactivated then this method must wait for
+   *    activation or deactivation to complete before continuing. If this does not occur in a reasonable
+   *    time, a std::runtime_error exception is thrown to indicate this bundle was unable to be stopped.
+   * -# If this bundle was already stopped, then this method returns immediately.
+   * -# A bundle event of type BundleEvent::STOPPING is fired.
+   * -# The BundleActivator::Stop(BundleContext) method of this bundle's BundleActivator, if one is specified,
+   *    is called. If that method throws an exception, this method must continue to stop this bundle
+   *    and a std::runtime_error exception must be thrown after completion of the remaining steps.
+   * -# %Any services registered by this bundle must be unregistered.
+   * -# %Any services used by this bundle must be released.
+   * -# %Any listeners registered by this bundle must be removed.
+   * -# A bundle event of type BundleEvent::STOPPED is fired.
+   *
+   * @throws std::runtime_error If the bundle failed to stop.
+   */
+  virtual void Stop();
+
+  /**
+   * Uninstalls this bundle.
+   * 
+   * This method causes the Framework to notify other bundles that this bundle is being uninstalled,
+   * and then uninstalls this bundle. The Framework must remove any resources related to this bundle 
+   * that it is able to remove.
+   *
+   * The following steps are required to uninstall a bundle:
+   * -# This bundle is stopped as described in the Bundle.Stop method.
+   * -# A bundle event of BundleEvent::UNINSTALLED is fired.
+   * -# This bundle and any persistent storage area provided for this bundle by the Framework are removed.
+   *
+   * @throws std::runtime_error If the bundle could not be uninstalled.
   *
+   */
+  virtual void Uninstall();
+
+private:
+
+  friend class CoreModuleActivator;
+  friend class ModuleRegistry;
+  friend class ServiceReferencePrivate;
+  friend class Framework;
+
+  ModulePrivate* d;
+
+  Module();
+
+  void Init(CoreModuleContext* coreCtx, ModuleInfo* info);
+  void Uninit();
+
+protected:
+  // purposely not implemented

```

Comment:
There is no reason. This may have happened during development and I forgot to move them back under ````private````. Good catch.